R-packages/evalcast/R/geo_type_selector.R

```diff
@@ -0,0 +1,50 @@
+
+geo_type_selector <- function(geo_type, predictions_cards) {
+  # tries to determine which geo_type we want and/or validate
+  # our selection based on the content of predictions_cards
+  # 
+  # Wraps the function below for ease of testing (we test the validator,
+  # rather than this function)
+  pred_card_types <- predictions_cards %>%
+    transmute(type_len = nchar(.data$geo_value),
+              has_chars = str_detect(.data$geo_value, "[aA-zZ]")) %>%
+    distinct()
+  geo_type <- match.arg(geo_type, c("county", "hrr", "msa", "dma", "state"))
```

Comment:
This should be the first line so we quit without doing anything if the wrong type is specified.